### PR TITLE
Reader: move reader-infinite-stream into client/reader

### PR DESCRIPTION
**Pull request by bluefuton**
#### Changes proposed in this Pull Request

A number of Reader-specific components and blocks currently live in the main components and blocks directories, even though they are not used elsewhere in Calypso. 

As discussed with @blowery, I feel that these components and blocks should live inside `client/reader` if they are not likely to be reused outside Reader.

#### Testing instructions

Load Following Manage at http://calypso.localhost:3000/following/manage and try scrolling through your subscriptions.

### CODE REVIEW

**Comment by jsnajdr** [comment]
One more nit about the `reader-infinite-stream/index.js` file: it contains JSX markup so technically it should be `.jsx`. Matters very little, but now when we're renaming it anyway...

File: client/reader/following-manage/feed-search-results.jsx [hunk 1]
```diff
@@ -16,8 +16,8 @@ import classnames from 'classnames';
 import Button from 'components/button';
 import ReaderSubscriptionListItemPlaceholder from 'blocks/reader-subscription-list-item/placeholder';
 import { READER_FOLLOWING_MANAGE_SEARCH_RESULT } from 'reader/follow-sources';
-import InfiniteStream from 'components/reader-infinite-stream';
-import { siteRowRenderer } from 'components/reader-infinite-stream/row-renderers';
+import InfiniteStream from 'reader/components/reader-infinite-stream';
+import { siteRowRenderer } from 'reader/components/reader-infinite-stream/row-renderers';
```
[the remark this answers is not in the document]
Comment (bluefuton) [reply client/reader/following-manage/feed-search-results.jsx]: That's true. The flat list feels a bit messy to me - some of the directories contain a controller and correspond to a route, and some contain just a standalone component. I was thinking of having `reader/blocks`, `reader/components` and keeping the existing `reader/lib` to organise things a bit.

**Comment by bluefuton** [comment]
> I found another two Reader-only components while reviewing:
> `blocks/reader-subscription-list-item`
> `blocks/reader-recommended-sites`
> Would they be also good candidates for moving?

Indeed they would. I'll move those in followup PRs. We also have these prime candidates:

* `components/reader-main`
* `components/reader-popover`

File: client/reader/following-manage/feed-search-results.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/reader/following-manage/feed-search-results.jsx]: > I was thinking of having reader/blocks, reader/components and keeping the existing reader/lib to organise things a bit.

That sounds like a plan :+1: `reader/components/reader-infinite-stream` makes sense then.

**Pull request merged**